compiler/rustc_resolve/src/late.rs

```diff
@@ -4321,7 +4321,7 @@ impl<'a: 'ast, 'b, 'ast, 'tcx> LateResolutionVisitor<'a, 'b, 'ast, 'tcx> {
                 );
             }
             ExprKind::Closure(..) => visit::walk_expr(self, expr),
-            ExprKind::Async(..) => {
+            ExprKind::Gen(..) => {
                 self.with_label_rib(RibKind::ClosureOrAsync, |this| visit::walk_expr(this, expr));
```

Comment:
Rib could be renamed

Reply:
surprisingly this `RibKind` is also used for functions to allow them access to generics of their surrounding scope in case they are methods. I did not look too deeply into it, but there may be some opportunity for cleanups or even hardening of the logic